src/js/components/stock-movement/inbound/StockMovementInboundFilters.jsx

```diff
@@ -0,0 +1,209 @@
+    },
+    getDynamicAttr: ({
+      fetchLocations,
+    }) => ({
+      loadOptions: fetchLocations,
+    }),
+  },
+  destination: {
+    type: FilterSelectField,
+    attributes: {
+      valueKey: 'id',
+      filterElement: true,
+      placeholder: 'Destination',
+      showLabelTooltip: true,
+      disabled: true,
+    },
+  },
+  requestedBy: {
+    type: FilterSelectField,
+    attributes: {
+      async: true,
+      openOnClick: false,
+      autoload: false,
+      cache: false,
+      valueKey: 'id',
+      labelKey: 'name',
+      options: [],
+      filterOptions: options => options,
+      filterElement: true,
+      placeholder: 'Requested By',
+      showLabelTooltip: true,
+    },
+    getDynamicAttr: ({
+      fetchUsers,
+    }) => ({
+      loadOptions: fetchUsers,
+    }),
+  },
+  createdBy: {
+    type: FilterSelectField,
+    attributes: {
+      async: true,
+      openOnClick: false,
+      autoload: false,
+      cache: false,
+      valueKey: 'id',
+      labelKey: 'name',
+      options: [],
+      filterOptions: options => options,
+      filterElement: true,
+      placeholder: 'Created By',
+      showLabelTooltip: true,
+    },
+    getDynamicAttr: ({
+      fetchUsers,
+    }) => ({
+      loadOptions: fetchUsers,
+    }),
+  },
+  updatedBy: {
+    type: FilterSelectField,
+    attributes: {
+      async: true,
+      openOnClick: false,
+      autoload: false,
+      cache: false,
+      valueKey: 'id',
+      labelKey: 'name',
+      options: [],
+      filterOptions: options => options,
+      filterElement: true,
+      placeholder: 'Updated By',
+      showLabelTooltip: true,
+    },
+    getDynamicAttr: ({
+      fetchUsers,
+    }) => ({
+      loadOptions: fetchUsers,
+    }),
+  },
+  createdAfter: {
+    type: DateFilter,
+    attributes: {
+      label: 'react.stockMovement.filter.createdAfter.label',
+      defaultMessage: 'Created after',
+      dateFormat: 'MM/DD/YYYY',
+      filterElement: true,
+    },
+  },
+  createdBefore: {
+    type: DateFilter,
+    attributes: {
+      label: 'react.stockMovement.filter.createdBefore.label',
+      defaultMessage: 'Created before',
+      dateFormat: 'MM/DD/YYYY',
+      filterElement: true,
+    },
+  },
+};
+
+const StockMovementInboundFilters = ({
+  setFilterParams,
+  debounceTime,
+  minSearchLength,
+  filterParams,
+  fetchStatuses,
+  shipmentStatuses,
+  isShipmentStatusesFetched,
+}) => {
+  const fetchUsers = debounceUsersFetch(debounceTime, minSearchLength);
+  const fetchLocations = debounceLocationsFetch(debounceTime, minSearchLength, [], true);
+
+
+  useEffect(() => {
+    if (!isShipmentStatusesFetched) fetchStatuses();
+  }, []);
+
+  return (
+    <div className="d-flex flex-column list-page-filters">
+      <FilterForm
+        searchFieldId="q"
+        searchFieldPlaceholder="Search by order number of description"
+        filterFields={filterFields}
+        defaultValues={{
+          ...filterParams,
```

Comment:
you could just make defaultValues as
 ```
destination: {
      id: currentLocation.id,
      value: currentLocation.id,
      name: currentLocation.name,
      label: currentLocation.name,
    },
```
and you wouldn't have to do this `onClear` prop. Assigning destination to `filterParams` as default state disables you from assigning it again if the `filterParams` change, so I probably know, why you made this `onClear`, but if you had passed `destination` as `defaultValues` here, you wouldn't have had to do this `onClear`.